src/index.ts

```diff
@@ -163,6 +163,34 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
     }
   })()
 
+  // formats the etag depending on the response context. if the entityId
+  // is invalid, returns an empty string (instead of null) to prevent the
+  // the potentially disastrous scenario where the value of the Etag resp
+  // header is "null". Could be modified in future to base64 encode etc
+  const formatETag = (entityId: any = pathKey, validatorType: string = 'strong') => {
+    if (!entityId) {
+      return ''
+    }
+    switch (validatorType) {
+      case 'weak':
+        if (!entityId.startsWith('W/')) {
+          return `W/${entityId}`
+        }
+        console.log('weakened', entityId)
```

Comment:
nitpick: This debug log should be removed, as it'll create a lot of noise on large production sites.

Reply:
Fixed with aa980fa